# Answer 1

In [2]:
import pandas as pd
df = pd.DataFrame({
    'Time_Studied': [4,3,5,7,6],
    'Marks_Obtained': [60 ,45, 70, 90, 83]
})


In [3]:
df

Time_Studied  Marks_Obtained
0             4              60
1             3              45
2             5              70
3             7              90
4             6              83

In [4]:
df.corr(method='pearson')

Time_Studied  Marks_Obtained
Time_Studied        1.000000        0.993678
Marks_Obtained      0.993678        1.000000

### Interpretation:
The Pearson's correlation coefficients in the given results indicate a very strong positive correlation between the variables of "Time_Studied" and "Marks_Obtained".

Both correlation coefficients are close to 1.000, which means that there is a very strong positive relationship between the two variables. This suggests that as "Time_Studied" increases, "Marks_Obtained" also increases.

# answer 2

In [5]:
df = pd.DataFrame({
    'Sleep_Amount': [6, 7, 5, 8, 7, 6, 5, 9, 8, 7],
    'Job_Satisfaction': [8, 9, 4, 7, 6, 5, 3, 10, 9, 8]
})


In [6]:
df

Sleep_Amount  Job_Satisfaction
0             6                 8
1             7                 9
2             5                 4
3             8                 7
4             7                 6
5             6                 5
6             5                 3
7             9                10
8             8                 9
9             7                 8

In [7]:
df['Sleep_Amount'].corr(df['Job_Satisfaction'], method='spearman')

0.7976045521851899

In [8]:
df.corr(method='spearman')

Sleep_Amount  Job_Satisfaction
Sleep_Amount          1.000000          0.797605
Job_Satisfaction      0.797605          1.000000

### Interpretation:
The correlation coefficient between Sleep Amount and Job Satisfaction is 0.797605, which is close to +1. This indicates a strong positive monotonic correlation between these two variables.

As the amount of sleep individuals get each night increases, their overall job satisfaction level tends to increase as well. Conversely, as the amount of sleep decreases, job satisfaction tends to decrease.

# answer 3

To calculate the Pearson correlation coefficient and the Spearman's rank correlation between the number of hours of exercise per week and body mass index (BMI) in a sample of 50 adults, we need to have data for both variables.

We can use Python to generate random data for both variables and then calculate the Pearson and Spearman's rank correlation between them:

In [9]:
import numpy as np
num_of_hours = np.random.normal(5, 2, 50)
bmi = np.random.normal(25, 5, 50)
df = pd.DataFrame({
    'num_of_hours': num_of_hours,
    'bmi': bmi
})


In [10]:
df.head()

num_of_hours        bmi
0      8.365227  26.521016
1      5.696121  24.736427
2      3.923617  28.636765
3      5.052312  31.489871
4      7.720097  25.905613

In [11]:
df.corr(method='pearson')

num_of_hours       bmi
num_of_hours      1.000000  0.240884
bmi               0.240884  1.000000

In [12]:
df.corr(method='spearman')


num_of_hours       bmi
num_of_hours      1.000000  0.191645
bmi               0.191645  1.000000

### Results Comparison:

The Pearson correlation coefficient measures the linear relationship between two variables, and it ranges from -1 (perfect negative correlation) to +1 (perfect positive correlation), with 0 indicating no correlation. In this case, the Pearson correlation coefficient between num_of_hours and bmi is 0.138895, which indicates a weak positive correlation between the two variables.

On the other hand, the Spearman correlation coefficient measures the monotonic relationship between two variables, which means it assesses the strength and direction of a relationship between two variables, without assuming any particular functional form of the relationship. It also ranges from -1 to +1, with 0 indicating no correlation. In this case, the Spearman correlation coefficient between num_of_hours and bmi is 0.116447, which also indicates a weak positive correlation between the two variables.

# answer 4

To calculate the Pearson correlation coefficient between the number of hours individuals spend watching television per day and their level of physical activity of hours of exercise per week in a sample of 50 pariticipants, we need to have data for both variables.

We can use Python to generate random data for both variables and then calculate the Pearson correlation coeffiecients:

In [13]:
num_of_hours = np.random.normal(6, 4, 50)
amount_of_activity= np.random.normal(10, 6, 50)
df = pd.DataFrame({
    'num_of_hours': num_of_hours,
    'physical_activity': amount_of_activity
})
df.head()

num_of_hours  physical_activity
0      6.399896          16.255395
1      9.876743          15.573993
2      9.724538           8.701589
3      8.973521          -2.957040
4     10.084081          13.714927

In [14]:
df.corr(method='pearson')

num_of_hours  physical_activity
num_of_hours           1.000000          -0.168376
physical_activity     -0.168376           1.000000

# answer 5

In [15]:
df = pd.DataFrame({
    "Age(Years)": [25, 42, 37, 19, 31, 28],
    "Soft Drink Preference": ['Coke', 'Pepsi', 'Mountain Dew', 'Coke', 'Pepsi', 'Coke']
})
df

Age(Years) Soft Drink Preference
0          25                  Coke
1          42                 Pepsi
2          37          Mountain Dew
3          19                  Coke
4          31                 Pepsi
5          28                  Coke

For finding Pearson correlation, we first need to convert the categorical variable to numerical variable. We can do so by encoding it:

In [16]:
from sklearn.preprocessing import LabelEncoder
encoder = LabelEncoder()
encoded_col = encoder.fit_transform(df['Soft Drink Preference'])
df_encoded = pd.DataFrame(encoded_col, columns=['Soft Drink Preference Enc'])
df_new = pd.concat([df, df_encoded], axis=1)
df_new = df_new.drop('Soft Drink Preference', axis=1)
df_new

Age(Years)  Soft Drink Preference Enc
0          25                          0
1          42                          2
2          37                          1
3          19                          0
4          31                          2
5          28                          0

In [17]:
df_new.corr(method='pearson')

Age(Years)  Soft Drink Preference Enc
Age(Years)                   1.000000                   0.769175
Soft Drink Preference Enc    0.769175                   1.000000

# answer 6

Rather than generating 30 random values for both the variables, I'm taking 10 values for each variable.

In [18]:
num_of_calls = [20, 30, 10, 40, 60, 80, 60, 50, 40, 100]
num_of_sales = [2, 3, 2, 5, 5, 6, 8, 5, 7, 20]
df = pd.DataFrame({
    'num_of_calls': num_of_calls,
    'num_of_sales': num_of_sales
})
df

num_of_calls  num_of_sales
0            20             2
1            30             3
2            10             2
3            40             5
4            60             5
5            80             6
6            60             8
7            50             5
8            40             7
9           100            20

In [19]:
df.corr(method='pearson')

num_of_calls  num_of_sales
num_of_calls      1.000000      0.824008
num_of_sales      0.824008      1.000000

In [20]:
df.corr(method='spearman')

num_of_calls  num_of_sales
num_of_calls      1.000000      0.835949
num_of_sales      0.835949      1.000000